fault.h

```diff
@@ -89,6 +133,143 @@ struct kibosh_fault_read_delay {
     double fraction;
 };
 
+/**
+ * The class for Kibosh faults that make files unwritable.
+ */
+struct kibosh_fault_unwritable {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+     * The path suffix, can be used to specify a file extension.
+     */
+    char *suffix;
+
+    /**
+     * The error code to return from write faults.
+     */
+    int code;
+};
+
+/**
+ * The class for Kibosh faults that lead to delays when writing.
+ */
+struct kibosh_fault_write_delay {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+     * The path suffix, can be used to specify a file extension.
+     */
+    char *suffix;
+
+    /**
+     * The number of milliseconds to delay the read.
+     */
+    uint32_t delay_ms;
+
+    /**
+     * The fraction of writes that are delayed.
+     */
+    double fraction;
+};
+
+/**
+ * The class for Kibosh faults that lead to corrupted data when reading.
+ */
+struct kibosh_fault_read_corrupt {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+     * The path suffix, can be used to specify a file extension.
+     */
+    char *suffix;
+
+    /**
+     * Mode of byte corruption.
+     * 1000 -> zeros (default)
+     * 1001 -> random values
+     * 1100 -> sequential zeros
+     * 1101 -> sequential random values
+     * 1200 -> silently drop a fraction of bytes at the end of buffer
+     */
+    int mode;
+
+    /**
+     * Number of corruption fault injected before switching to unwritable fault.
+     * Less than 0 means never switch.
+     */
+    int count;
+
+    /**
+     * The fraction of bytes to be corrupted.
+     */
+    double fraction;
+};
+
+/**
+ * The class for Kibosh faults that lead to corrupted data when writing.
+ */
+struct kibosh_fault_write_corrupt {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+    * The path suffix, can be used to specify a file extension.
+    */
+    char *suffix;
+
+    /**
+     * Mode of byte corruption.
+     * 1000 -> zeros (default)
+     * 1001 -> random values
+     * 1100 -> sequential zeros
+     * 1101 -> sequential random values
+     * 1200 -> silently drop a fraction of bytes at the end of buffer
+     */
+    int mode;
```

Comment:
should use enum type defined earlier right

Reply:
Edited as suggested.